airstack-modules/modules/airstack/domain-name/domain-name.ts

```diff
@@ -0,0 +1,1074 @@
+import {
+  BigInt,
+  dataSource,
+  Bytes,
+  crypto,
+  log,
+  ByteArray,
+  ens,
+} from "@graphprotocol/graph-ts";
+
+import {
+  AirAccount,
+  AirEntityCounter,
+  AirToken,
+  AirBlock,
+  AirMeta,
+  AirDomain,
+  AirDomainOwnerChangedTransaction,
+  AirDomainTransferTransaction,
+  AirResolver,
+  AirDomainNewResolverTransaction,
+  AirDomainNewTTLTransaction,
+  AirNameRegisteredTransaction,
+  AirNameRenewedTransaction,
+  AirAddrChanged,
+} from "../../../generated/schema";
+import { uint256ToByteArray, AIR_META_ID, AIR_DOMAIN_OWNER_CHANGED_ENTITY_COUNTER_ID, AIR_ADDR_CHANGED_TRANSACTION_COUNTER_ID, AIR_NAME_RENEWED_TRANSACTION_COUNTER_ID, AIR_NAME_REGISTERED_TRANSACTION_COUNTER_ID, AIR_DOMAIN_NEW_TTL_TRANSACTION_COUNTER_ID, AIR_DOMAIN_NEW_RESOLVER_ENTITY_COUNTER_ID, AIR_DOMAIN_TRANSFER_ENTITY_COUNTER_ID, BIGINT_ONE, SUBGRAPH_SCHEMA_VERSION, SUBGRAPH_VERSION, SUBGRAPH_NAME, SUBGRAPH_SLUG, processNetwork, BIG_INT_ZERO, ROOT_NODE, ZERO_ADDRESS, EMPTY_STRING } from "./utils";
+
+export namespace domain {
+  /**
+   * @dev This function tracks a domain owner change transaction
+   * @param blockHeight block number in the chain
+   * @param blockHash block hash
+   * @param blockTimestamp block timestamp
+   * @param logIndex txn log index
+   * @param chainId chain id
+   * @param newOwner specifies the new owner of the domain
+   * @param transactionHash transaction hash
+   * @param isMigrated specifies if the domain is migrated
+   * @param node specifies the node of the domain
+   * @param label specifies the label of the domain
+   * @param tokenAddress contract address of nft token
+   * @param fromOldRegistry specifies if the event is from the old registry
+   */
+  export function trackDomainOwnerChangedTransaction(
+    blockHeight: BigInt,
+    blockHash: string,
+    blockTimestamp: BigInt,
+    logIndex: BigInt,
+    chainId: string,
+    newOwner: string,
+    transactionHash: Bytes,
+    isMigrated: boolean,
+    node: Bytes,
+    label: Bytes,
+    tokenAddress: string,
+    fromOldRegistry: boolean,
+  ): void {
+    let block = getOrCreateAirBlock(chainId, blockHeight, blockHash, blockTimestamp);
+    let domainId = createAirDomainEntityId(node, label);
+    let domain = getOrCreateAirDomain(new Domain(domainId, chainId, block, tokenAddress));
+    if (fromOldRegistry && domain.isMigrated == true) {
+      // this domain was migrated from the old registry, so we don't need to hanlde old registry event now
```

Comment:
Please specify the reason. 
As a reviewer I am not sure on why

If we are ignoring the events form old registry, then check this at the first place at line 59

Reply:
also, its being checked first thing